src/Avalonia.FuncUI.ControlCatalog/Views/Tabs/MaskedTextBoxDemo.fs

```diff
@@ -0,0 +1,72 @@
+namespace Avalonia.FuncUI.ControlCatalog.Views
+
+open System
+open Avalonia.Controls
+open Elmish
+open Avalonia.FuncUI
+open Avalonia.FuncUI.DSL
+open Avalonia.FuncUI.Components
+open Avalonia.FuncUI.Elmish
+open Avalonia.Input
+open Avalonia.Interactivity
+open Avalonia.Layout
+
+module MaskedTextBoxDemo =
+    type State = { count: int }
+
+    let init = { count = 0 }
+
+    type Msg =
+    | SetCount of int
+
+    let update (msg: Msg) (state: State) : State =
+        match msg with
+        | SetCount number -> { state with count = number }
+           
+    let view (state: State) (dispatch) =
+        StackPanel.create [
+            StackPanel.spacing 10.0
+            StackPanel.children [
+                TextBox.create [
+                    TextBox.text (string state.count)
+                    TextBox.onTextChanged (fun text ->
+                        let isNumber, number = Int32.TryParse text
+                        if isNumber then
+                            dispatch (SetCount number)
+                            
+                    , SubPatchOptions.Never) // <- this is annoying.
+                                             // We should have this optional with the default being `Never`
+                    
+                    TextBox.useEffect (fun control ->
+                        
+                        let action = EventHandler<TextInputEventArgs>(fun sender args ->
+                            let isNumber, _ = Int32.TryParse args.Text
+                            if not isNumber then
+                                args.Text <- String.Empty
+                        )
+                            
+                        (TextBox.TextInputEvent, action, RoutingStrategies.Tunnel, false)
+                        |> control.AddHandler
+                        |> ignore
```

Comment:
this kind of usage could lead to memory leaks if it goes unchecked right? It would be nice to add something in the docs about it

Reply:
Yes, this leaks like crazy :D 

It's just as a prove of concept - later this would be only done once, in the **$onCreate**. This was just a test to see if this would actually work in order to bypass default event routing   and.. it does 😁

Reply:
(where in this case we basically intercept events in the `Tunneling` phase)